### PR TITLE
Additional notes on using vscode for workerd development

### PR DESCRIPTION
**Pull request by ohodson**
Adds a list of recommended extensions and a few tips.

Adds an `extensions.json` file that can be used to install the recommended extensions with one command.

### CODE REVIEW

**Review by jasnell — APPROVED** [review]

File: docs/vscode.md [hunk 1]
```diff
@@ -1,37 +1,27 @@
...
 
-```sh
-$ bazel clean --expunge
-$ bazel test //...
-```
+* [GitLens extension](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) for super charged git functionality within Visual Studio Code.
 
-There is an issue between workerd's bazel setup and Hedron's compile_commands.json generator (tracked in
-https://github.com/cloudflare/workerd/issues/506).
+You can install all of these extensions with the **Extensions: Configure Recommended Extensions (Workspace Folder)** command. You can do find this through the Visual Studio Code Command Palette (`shift+ctrl+p` on Linux / Windows, `shift+cmd+p` on OS X) and typing "Configure Recommended Extensions". The recommendations that will be installed can be found in the [.vscode/extensions.json](../.vscode/extensions.json) file.
```
Comment (jp4a50) [inline docs/vscode.md:22]: [nit] `You can do` -> `You can`

**Review by jp4a50 — APPROVED** [review]
Awesome! Appreciate you taking the time to improve this.

File: docs/vscode.md (hunk 1 above)
Comment (ohodson) [reply docs/vscode.md:22]: Thanks, experimenting with incorporating this comment in vscode...
```suggestion
You can install all of these extensions with the **Extensions: Configure Recommended Extensions (Workspace Folder)** command. You can find this through the Visual Studio Code Command Palette (`shift+ctrl+p` on Linux / Windows, `shift+cmd+p` on OS X) and typing "Configure Recommended Extensions". The recommendations that will be installed can be found in the [.vscode/extensions.json](../.vscode/extensions.json) file.
```

Comment (ohodson) [reply docs/vscode.md]: Thanks incorporated in https://github.com/cloudflare/workerd/pull/610/commits/60af28eff87c7ce4efed0cb08eb5a81366adfc70

**Pull request merged**